lib/modules/dosomething/dosomething_northstar/dosomething_northstar.module

```diff
@@ -43,6 +43,30 @@ function dosomething_northstar_libraries_info() {
   return $libraries;
 }
 
+
+/**
+ * Verify the given credentials against Northstar, returning the authorized
+ * user account if one exists.
+ *
+ * @param array $credentials
+ * @return mixed|null
+ */
+function dosomething_northstar_verify_user($credentials) {
+  $client = _dosomething_northstar_build_http_client();
+
+  $response = drupal_http_request($client['base_url'] . '/auth/verify', [
+    'headers' => $client['headers'],
+    'method' => 'POST',
+    'data' => json_encode($credentials),
+  ]);
+
+  if($response->code !== '200') {
+    return null;
```

Comment:
The only time this _should_ happen is if the given credentials doesn't exist in Northstar, which will return a `401`. It might be worth explicitly checking for the `401` to return `NULL` and then triggering a watchdog if any other code is returned?